src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -1047,3 +1052,47 @@ void ControllerScriptInterfaceLegacy::softStart(int deck, bool activate, double
     // activate the ramping in scratchProcess()
     m_ramp[deck] = true;
 }
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const ControllerScriptInterfaceLegacy::WellKnownCharsets targetCharset,
+        const QString& value) {
+    switch (targetCharset) {
+    case WellKnownCharsets::Latin1:
+    case WellKnownCharsets::ISO_8859_1:
+        return convertCharset(QStringLiteral("ISO-8859-1"), value);
+    case WellKnownCharsets::Latin9:
+    case WellKnownCharsets::ISO_8859_15:
+        return convertCharset(QStringLiteral("ISO-8859-15"), value);
+    case WellKnownCharsets::UCS2:
+    case WellKnownCharsets::ISO_10646_UCS_2:
+        return convertCharset(QStringLiteral("ISO-10646-UCS-2"), value);
+    default:
+        m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unknown charset specified"));
+        return QByteArray();
+    }
+}
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const QString& targetCharset, const QString& value) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+    auto* pCodec = QTextCodec::codecForName(targetCharset.toUtf8());
+    if (!pCodec) {
+        m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unable to open encoder"));
+        return QByteArray();
+    }
+    return pCodec->makeEncoder(QTextCodec::Flag::ConvertInvalidToNull)->fromUnicode(value);
+#else
+#if QT_VERSION > QT_VERSION_CHECK(6, 8, 0)
+    const auto* const encoderName = targetCharset.data();
+#else
+    auto* const encoderName = targetCharset.toUtf8().data();
+#endif
+    QStringEncoder fromUtf16 = QStringEncoder(
+            encoderName, QStringEncoder::Flag::ConvertInvalidToNull);
```

Comment:
```suggestion
    QStringEncoder fromUtf16 = QStringEncoder(
            QAnyStringView(targetCharset), QStringEncoder::Flag::ConvertInvalidToNull);
    if (!fromUtf16.isValid()) {
        m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unable to open encoder"));
        return QByteArray();
    }
    return fromUtf16(value);
```

@JoergAtGithub I'm not sure what I'm doing here, but would that code work for every Qt>6?